tests/yamitranscode.cpp

```diff
@@ -244,11 +244,11 @@ SharedPtr<VppOutput> createOutput(TranscodeParams& para, const SharedPtr<VADispl
     return output;
 }
 
-SharedPtr<FrameAllocator> createAllocator(const SharedPtr<VppOutput>& output, const SharedPtr<VADisplay>& display)
+SharedPtr<FrameAllocator> createAllocator(const SharedPtr<VppOutput>& output, const SharedPtr<VADisplay>& display, int32_t extraSize)
 {
     uint32_t fourcc;
     int width, height;
-    SharedPtr<FrameAllocator> allocator(new PooledFrameAllocator(display, 5));
+    SharedPtr<FrameAllocator> allocator(new PooledFrameAllocator(display, std::max(extraSize, 5)));
```

Comment:
Two types of surface included in encoder. One for input yuv, one for reconstruct yuv. Encoder can decide the size of reconstruct yuv list. However, for input yuv, we handle it in two ways:
1.  If you use `virtual YamiStatus encode(VideoFrameRawData* frame) = 0;`, we create a new surface and copy content for you.
2. If you use `virtual YamiStatus encode(const SharedPtr<VideoFrame>& frame) = 0;`, we will simply put the frame to reorder list. This mode is for zero copy encode, you can think the frame is from camera or egl texture. 
This patch fix issue for scenario 2, suppose you want set the encode sequence as IBBBBP, you need allocate at least 6 frames for input surface list, because we will cache all B frames until we have P frame.